### PR TITLE
basic feature implementation of app repo client

### PR DESCRIPTION
**Pull request by empratyush**
Please. don't merge this it's not ready yet

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
         </activity>
+
+        <receiver android:name=".utils.APKInstallReceiver"/>
+
+        <service android:name=".service.KeepAppActive"
+            android:foregroundServiceType="dataSync"/>
     </application>
 
-</manifest>
+</manifest>
```
Comment (flawedworld) [inline app/src/main/AndroidManifest.xml:40]: add newline

Comment (empratyush) [reply app/src/main/AndroidManifest.xml:41]: Okay fixed

**Comment by empratyush** [comment]
#19 client side stuff  ✔️

**Comment by empratyush** [comment]
it's almost ready now, need testing and review.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -57,15 +61,51 @@ android {
...
+            }
+        }
+    }
 }
 
 dependencies {
+    /*Default android*/
     implementation("org.jetbrains.kotlin:kotlin-stdlib:1.6.10")
     implementation("androidx.core:core-ktx:1.7.0")
     implementation("androidx.appcompat:appcompat:1.4.0")
```
Comment (Zanthed) [inline app/build.gradle.kts:81]: just FYI, appcompat has updated to 1.4.1 few days ago: https://developer.android.com/jetpack/androidx/releases/appcompat#1.4.1

Comment (empratyush) [reply app/build.gradle.kts]: updated.

**Pull request merged**